aten/src/ATen/native/Pool.h

```diff
@@ -58,6 +58,8 @@ template<typename T>
 static inline T pooling_output_shape(
       T inputSize, T kernelSize, T pad, T stride, T dilation, bool ceil_mode) {
     TORCH_CHECK(stride != 0, "stride should not be zero");
+    TORCH_CHECK(pad >= 0,
+                "pad must be non-negative, but got pad: ", pad);
```

Comment:
Looking at the issue description, should we also check that `pad <= kernel_size/2` and update the message accordingly?

Reply:
Yep, added check! Thanks